Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py

```diff
@@ -438,46 +474,97 @@ def fetch_indicators_command(client,
-                        client.feed_url_to_config.get(url, {}).get('indicator_type'), itype, auto_detect, value)
-                    indicator_data = {
-                        'value': value,
-                        'type': indicator_type,
-                        'rawJSON': attributes,
-                    }
-                    if enrichment_excluded:
-                        indicator_data['enrichmentExcluded'] = enrichment_excluded
-
-                    if (create_relationships
-                            and client.feed_url_to_config.get(url, {}).get('relationship_name')
-                            and attributes.get('relationship_entity_b')
-                        ):
-                        relationships_lst = EntityRelationship(
-                            name=client.feed_url_to_config.get(url, {}).get('relationship_name'),
-                            entity_a=value,
-                            entity_a_type=indicator_type,
-                            entity_b=attributes.get('relationship_entity_b'),
-                            entity_b_type=FeedIndicatorType.indicator_type_by_server_version(
-                                client.feed_url_to_config.get(url, {}).get('relationship_entity_b_type')),
-                        )
-                        relationships_of_indicator = [relationships_lst.to_indicator()]
-                        indicator_data['relationships'] = relationships_of_indicator
-
-                    if len(client.custom_fields_mapping.keys()) > 0 or TAGS in attributes:
-                        custom_fields = client.custom_fields_creator(attributes)
-                        indicator_data["fields"] = custom_fields
-
-                    indicators.append(indicator_data)
+                attributes, indicator_values = get_indicator_fields(line, url, feed_tags, tlp_color, client)
+                demisto.debug(f"Got the following indicator values - {indicator_values}")
+
+                if indicator_values and len(indicator_values) == 1:
+                    indicators.append(process_indicator_data(
+                        client,
+                        indicator_values[0],
+                        attributes,
+                        url,
+                        itype,
+                        auto_detect,
+                        create_relationships,
+                        enrichment_excluded
+                    ))
+                elif indicator_values:
+                    for indicator_value in indicator_values:
+                        indicators.append(process_indicator_data(
+                            client,
+                            indicator_value,
+                            attributes,
+                            url,
+                            itype,
+                            auto_detect,
+                            create_relationships,
+                            enrichment_excluded
+                        ))
```

Comment:
The previous function can only return lists (empty or full) so no need to ask if indicator_values is defined and what is its length.
```suggestion
                    for indicator_value in indicator_values:
                        indicators.append(process_indicator_data(
                            client,
                            indicator_value,
                            attributes,
                            url,
                            itype,
                            auto_detect,
                            create_relationships,
                            enrichment_excluded
                        ))
```